src/models/poll.ts

```diff
@@ -172,4 +173,26 @@ export class Poll extends TypedEventEmitter<Exclude<PollEvent, PollEvent.New>, P
 
         this.emit(PollEvent.Responses, this.responses);
     }
+
+    private validateEndEvent(endEvent?: MatrixEvent): boolean {
+        if (!endEvent) {
+            return false;
+        }
+        /**
+         * Repeated end events are ignored -
+         * only the first (valid) closure event by origin_server_ts is counted.
+         */
+        if (this.endEvent && this.endEvent.getTs() < endEvent.getTs()) {
+            return false;
+        }
+
+        /**
+         * MSC3381
+         * If a m.poll.end event is received from someone other than the poll creator or user with permission to redact
+         * other's messages in the room, the event must be ignored by clients due to being invalid.
+         */
+        const roomCurrentState = this.room.currentState;
+        const endEventSender = endEvent.getSender();
+        return !!endEventSender && roomCurrentState.maySendRedactionForEvent(this.rootEvent, endEventSender);
```

Comment:
this doesn't seem to be quite the same thing as the comment above. It's possible that `endEventSender` is the same as the sender of the root event, and yet they lack permission to redact that create event (for example, because sending `m.room.redaction` events has been disabled).

Reply:
good catch, have added a `sender === creator` check here